torch/_inductor/compile_fx.py

```diff
@@ -158,59 +159,20 @@ def compile_fx_inner(
     is_inference=False,
     boxed_forward_device_index=None,
 ):
-    if is_tf32_warning_applicable(gm):
-        _warn_tf32_disabled()
-
-    if dynamo_utils.count_calls(gm.graph) == 0:
-        return make_boxed_func(gm.forward)
-
-    # lift the maximum depth of the Python interpreter stack
-    # to adapt large/deep models
-    sys.setrecursionlimit(max(sys.getrecursionlimit(), 2000))
-
-    _step_logger()(
-        logging.INFO,
-        "torchinductor compiling "
-        f"{'BACKWARDS' if is_backward else 'FORWARDS'} "
-        f"graph {graph_id}",
-    )
-    V.debug.fx_graph(gm, example_inputs)
-
-    if cudagraphs is None:
-        cudagraphs = config.triton.cudagraphs
-
-    shape_env = _shape_env_from_inputs(example_inputs)
-
-    fake_mode = detect_fake_mode(example_inputs)
-    if not fake_mode:
-        fake_mode = torch._subclasses.FakeTensorMode(allow_non_fake_inputs=True)
-        FakeTensorProp(gm, mode=fake_mode).propagate(*example_inputs)
-    else:
-        FakeTensorProp(gm, mode=fake_mode).propagate_dont_convert_inputs(
-            *example_inputs
-        )
-    # pattern matcher passes might not preserve striding information
-    # on node.meta["val"]. if in the future we rely on these being
-    # correct we will need to fix.
-
-    with V.set_fake_mode(fake_mode):
-        post_grad_passes(gm)
-        V.debug.fx_graph_transformed(gm, example_inputs)
+    # Inputs to fx_codegen_and_compile
+    graph_args = [gm, example_inputs]
+    graph_kwargs = {
+        "cudagraphs": cudagraphs,
+        "num_fixed": num_fixed,
+        "is_backward": is_backward,
+        "graph_id": graph_id,
+        "cpp_wrapper": cpp_wrapper,
+        "aot_mode": aot_mode,
+    }
+    compiled_graph: CompiledFxGraph = FxGraphCache.load(fx_codegen_and_compile, graph_args, graph_kwargs)
 
-    with V.set_fake_mode(fake_mode):
-        graph = GraphLowering(
-            gm,
-            shape_env=shape_env,
-            num_static_inputs=num_fixed,
-            graph_id=graph_id,
-            cpp_wrapper=cpp_wrapper,
-            aot_mode=aot_mode,
-        )
-        with V.set_graph_handler(graph):
-            graph.run(*example_inputs)
-            compiled_fn = graph.compile_to_fn()
```

Comment:
TracingContext: compilation process can generate guards, you need to make sure you reemit these guards when loading